x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/delete_rules_bulk_route.ts

```diff
@@ -45,7 +48,17 @@ export const createDeleteRulesBulkRoute: Hapi.ServerRoute = {
             id,
             ruleId,
           });
-
+          const ruleStatuses = await savedObjectsClient.find<
+            IRuleSavedAttributesSavedObjectAttributes
+          >({
+            type: ruleStatusSavedObjectType,
+            perPage: 6,
+            search: `${rule?.id}`,
```

Comment:
Avoid the `rule?.id` check here by just moving these blocks in the if check like so:


```ts
        try {
          const rule = await deleteRules({
            actionsClient,
            alertsClient,
            id,
            ruleId,
          });
          if (rule != null) {
            const ruleStatuses = await savedObjectsClient.find<
              IRuleSavedAttributesSavedObjectAttributes
            >({
              type: ruleStatusSavedObjectType,
              perPage: 6,
              search: `${rule.id}`,
              searchFields: ['alertId'],
            });
            ruleStatuses.saved_objects.forEach(async obj =>
              savedObjectsClient.delete(ruleStatusSavedObjectType, obj.id)
            );
            return transformOrBulkError(idOrRuleIdOrUnknown, rule);
          } else {
            return getIdBulkError({ id, ruleId });
          }
        } catch (err) {
          return transformBulkError(idOrRuleIdOrUnknown, err);
        }
```

Change these three:
* delete_rules_bulk_route.ts
* delete_rules_route.ts
* read_rules_route.ts

You don't want to be sending searches of `null` down but instead skip the extra query altogether.